lib/features.js

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const hash = require('sha1')
```

Comment:
Worthwhile-sounding non-cryptographic contenders incude:

* https://www.npmjs.com/package/xxhash
* https://www.npmjs.com/package/farmhash

Both of them include an option to hash with a "seed", which maps really nicely to the way we incorporate the feature name into the hash.

It's worth a bit of time to find a good option here, but if we spend a week bikeshedding the details, we're doing something wrong :-).  @philbooth please feel fully empowered to pick an approach you like and run with it to get this PR merged.